custom_components/tuya_local/devices/holman_wx8_irrigation_controller.yaml

```diff
@@ -0,0 +1,208 @@
+name: Holman WX8
+products:
+  - id: 3c8bxnseywh7radc
+    model: WX8
+    manufacturer: Holman
+primary_entity:
+  entity: sensor
+  name: Watering
+  dps:
+    - id: 123
+      type: boolean
+secondary_entities:
+  - entity: switch
+    name: Zone 1
+    dps:
+      - id: 102
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Zone 2
+    dps:
+      - id: 103
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Zone 3
+    dps:
+      - id: 104
+        type: boolean
+        name: switch  
+  - entity: switch
+    name: Zone 4
+    dps:
+      - id: 104
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Zone 5
+    dps:
+      - id: 106
+        type: boolean
+        name: switch         
+  - entity: switch
+    name: Zone 6
+    dps:
+      - id: 107
+        type: boolean
+        name: switch       
+  - entity: switch
+    name: Zone 7
+    dps:
+      - id: 108
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Zone 8
+    dps:
+      - id: 112
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Start A
+    dps:
+      - id: 113
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Start B
+    dps:
+      - id: 114
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Start C
+    dps:
+      - id: 115
+        type: boolean
+        name: switch 
+  - entity: sensor
+    name: Firmware Version
+    category: diagnostic
+    dps:
+      - id: 116
+        name: sensor
+        type: integer     
+  - entity: sensor
+    name: Power Supply Voltage
+    category: diagnostic
+    dps:
+      - id: 117
+        name: sensor
+        type: integer
+        unit: "V"
+        mapping:
+          - scale: 10
+  - entity: sensor
+    name: Battery Voltage
+    category: diagnostic
+    dps:
+      - id: 118
+        type: integer
+        name: sensor
+        unit: "V"
+        mapping:
+          - scale: 10
+  - entity: sensor
+    name: Catchup Voltage
+    category: diagnostic
+    dps:
+      - id: 119
+        type: integer
+        name: sensor
+        unit: "V"
+        mapping:
+          - scale: 10
+  - entity: sensor
+    name: Device Flag
+    category: diagnostic
+    # dps:
+      # - id: 120
+        # type: bitfield
+        # name: sensor
+    dps:
+      - id: 120
+        optional: true
+        name: 9V Battery Good
```

Comment:
If you want to split this out to a separate sensor, this can be a binary_sensor with class: battery, but the logic is reversed (true means low battery, false means good).
If you need more debugging to figure out the masking and values, just add it as an attribute on the primary_entity without any masking.